lib/chainlock/chainlock.js

```diff
@@ -0,0 +1,278 @@
+  return doubleSha256(this.toBuffer()).reverse();
+};
+
+/**
+ * Computes the request ID for this ChainLock
+ * @returns {Buffer} - Request id for this chainlock
+ */
+ChainLock.prototype.getRequestId = function getRequestId() {
+  var bufferWriter = new BufferWriter();
+
+  var prefix = ChainLock.Constants.CLSIG_REQUESTID_PREFIX;
+  var prefixLength = prefix.length;
+
+  bufferWriter.writeVarintNum(prefixLength);
+  bufferWriter.write(Buffer.from(ChainLock.Constants.CLSIG_REQUESTID_PREFIX, 'utf-8'));
+  bufferWriter.writeUInt32LE(this.height);
+
+  // Double-sha is used to protect from extension attacks.
+  return doubleSha256(bufferWriter.toBuffer());
+};
+
+/**
+ * Computes signature id for a quorum entry
+ * @param {QuorumEntry} quorumEntry
+ * @returns {Buffer} - Signature id for this requestId and quorum.
+ */
+ChainLock.prototype.getSignatureIDForQuorumEntry = function (quorumEntry) {
+  var llmqType = quorumEntry.llmqType;
+  var quorumHash = quorumEntry.quorumHash;
+  var requestID = this.getRequestId();
+  var blockHash = this.blockHash;
+
+  var bufferWriter = new BufferWriter();
+  bufferWriter.writeUInt8(llmqType);
+  bufferWriter.write(Buffer.from(quorumHash, 'hex'));
+  bufferWriter.write(Buffer.from(requestID, 'hex'));
+  bufferWriter.write(blockHash);
+  return doubleSha256(bufferWriter.toBuffer());
+}
+
+/**
+ * Serializes chainlock to JSON
+ * @returns {Object} A plain object with the chainlock information
+ */
+ChainLock.prototype.toJSON = ChainLock.prototype.toObject = function toJSON() {
+  return {
+    height: this.height,
+    blockHash: BufferUtil.reverse(this.blockHash).toString('hex'),
+    signature: this.signature.toString('hex')
+  };
+};
+
+
+/**
+ * Serialize ChainLock
+ * @returns {string} - A hex encoded string of the chainlock
+ */
+ChainLock.prototype.toString = function toString() {
+  return this.toBuffer().toString('hex');
+};
+
+/**
+ * Serialize ChainLock to buffer
+ * @return {Buffer}
+ */
+ChainLock.prototype.toBuffer = function toBuffer() {
+  return this.toBufferWriter().concat();
+};
+
+/**
+ * @param {BufferWriter} bw - An existing instance BufferWriter
+ * @returns {BufferWriter} - An instance of BufferWriter representation of the ChainLock
+ */
+ChainLock.prototype.toBufferWriter = function toBufferWriter(bw) {
+  if (!bw) {
+    bw = new BufferWriter();
+  }
+  bw.writeInt32LE(this.height);
+  bw.write(this.blockHash);
+  bw.write(this.signature);
+  return bw;
+};
+/**
+ * Creates a copy of ChainLock
+ * @return {ChainLock} - a new copy instance of ChainLock
+ */
+ChainLock.prototype.copy = function copy() {
+  return ChainLock.fromBuffer(this.toBuffer());
+};
+
+/**
+ * Will return a string formatted for the console
+ *
+ * @returns {string} ChainLock block hash and height
+ */
+ChainLock.prototype.inspect = function () {
+  return '<ChainLock: ' + BufferUtil.reverse(this.blockHash).toString('hex') + ', height: ' + this.height + '>';
+};
+
+
+ChainLock.Constants = {
```

Comment:
Why not just `ChainLock.CLSIG_REQUESTID_PREFIX`?

Reply:
Followed the standard we have in other file, but I can do this too

Reply:
Fixed.